src/index.adoc

```diff
@@ -4,14 +4,570 @@ include::header.adoc[]
+  probably deserves a comment.
+- If it's for a longer discussion, schedule a call or just pick up the phone and call the other
+  person.
+- If you as a senior spend too much time correcting rookies' mistakes in juniors' codes, consider
+  changing the process: let juniors first peer-review the code and only then there's a PR with you
+  involved. Among other benefits, this'll elevate the cost for writing a shitty code, so people will
+  soon stop using you as an oracle and they start thinking for themselves.
+
+Finally, if it's not clear why the code works correctly, you shouldn't merge it. Author can use
+different ways to persuade you, she did a good job:
+
+- adding explanatory comments,
+- writing tests, demonstrating the desired functionality,
+- structuring a code in a way, that certain behavior is enforced by design,
+- Q: "hey, I'm not sure about this", A: "Me neither, but I've tested it manually and it seems to
+  work. It's poorly documented, so it'll do for now I guess."
+
+The code review process we usually use looks like this:
+
+- Reviewer comments to the PR.
+- Author reacts :thumbsup:, if he accepts the proposed change. If it's something more complex, the
+  discussion can take place. Tip: avoid writing comments for anything more complicated. It's
+  inefficient and frustrating. Have a quick call instead.
+- Reviewer checks that she likes the outcome by clicking "mark as resolved".
+- If reviewer has only cosmettic comments, she can approve the PR indicating he doesn't care that
+  much which suggestions author implements.
+
+Bad practices:
+
+- Leave the comment without any reaction.
+- Resolving the reviewer's comment by the author in a case where author is not 100% sure he fixed it
+  in a way reviewer intended.
+
+The changes are usually pushed as a new "fixing PR comments" commit to the feature branch. This
+allows the reviewer to see the edits in the fixing commit, or see the whole proposed change in the
+feature branch. For this to work, the feature branch shouldn't contain more than one logical change.
+Also, the "fixing PR comments" commits are not useful after the branch is merged--therefore the
+feature branch should be squashed to a small number of commits (often just one) before merging.
+
+Another approach is: you can put more than one atomic change into a branch. In such a case, one
+atomic change gets represented by one commit. The reviewer reviews the branch commit-by-commit and
+when she requires change, author edits the history. Reviewer doesn't see the correcting diff itself,
+but can then check out her comments, the current state of the code and see, if the code looks good
+now. Pros: it works better for doing several tasks simultaneously. For example, you start coding a
+feature, refactor something on-the-fly and fix a bug you discover. It's much easier to have just one
+branch for that. Cons: reviewer doesn't see an exact diff, that author produced to implement their
+change.
```

Comment:
> author edits the history. Reviewer doesn't see the correcting diff

A solution actually exists, `git range-diff`, though not many people seem to use it. I often find it useful in exactly this kind of situation: verifying some small change to a long commit series.